src/test/java/com/google/cloud/partners/pubsub/kafka/SubscriberImplTest.java

```diff
@@ -69,18 +82,20 @@
 import org.junit.After;
 import org.junit.Before;
 import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
 import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
 
 @RunWith(MockitoJUnitRunner.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING) // FIXME: Delete must happen before List
```

Comment:
It's been a while since I've been in this project, but looking at the way things are now, it seems like we already could have had issues if delete didn't happen before List. Could you just move the code that adds the SubscriptionProperties for the topics being delete to the test methods that exercise these actions?

Reply:
Thank you for reviewing this. I agree that strict ordering of unit tests should be avoided, however, I noticed the order is already important when I run `listSubscriptions` test in isolation:
```
java.lang.AssertionError: 
Expected :3
Actual   :4
```
this happens because `deleteSubscription` doesn't happen as you mentioned. adding the `FixMethodOrder` annotation just makes this explicit, but doesn't fix the issue. I will try to implement your suggestion and see what happens :)